src/ServiceControlInstaller.Engine/Configuration/ServiceControl/SettingsList.cs

```diff
@@ -5,37 +5,37 @@ namespace ServiceControlInstaller.Engine.Configuration.ServiceControl
     // See Compatibility.cs for version switching that isn't related to Settings
     public static class ServiceControlSettings
     {
-        public static SettingInfo VirtualDirectory = new SettingInfo { Name = "ServiceControl/VirtualDirectory" };
-        public static SettingInfo Port = new SettingInfo { Name = "ServiceControl/Port" };
-        public static SettingInfo DatabaseMaintenancePort = new SettingInfo
+        public static readonly SettingInfo VirtualDirectory = new SettingInfo { Name = "ServiceControl/VirtualDirectory" };
+        public static readonly SettingInfo Port = new SettingInfo { Name = "ServiceControl/Port" };
+        public static readonly SettingInfo DatabaseMaintenancePort = new SettingInfo
         {
             Name = "ServiceControl/DatabaseMaintenancePort",
             SupportedFrom = new Version(2, 0, 0)
         };
-        public static SettingInfo HostName = new SettingInfo { Name = "ServiceControl/HostName" };
-        public static SettingInfo LogPath = new SettingInfo { Name = "ServiceControl/LogPath" };
-        public static SettingInfo DBPath = new SettingInfo { Name = "ServiceControl/DBPath" };
-        public static SettingInfo ForwardAuditMessages = new SettingInfo
+        public static readonly SettingInfo HostName = new SettingInfo { Name = "ServiceControl/HostName" };
+        public static readonly SettingInfo LogPath = new SettingInfo { Name = "ServiceControl/LogPath" };
+        public static readonly SettingInfo DBPath = new SettingInfo { Name = "ServiceControl/DBPath" };
+        public static readonly SettingInfo ForwardAuditMessages = new SettingInfo
         {
             Name = "ServiceControl/ForwardAuditMessages",
             RemovedFrom = new Version(4, 0, 0)
         };
 
-        public static SettingInfo ForwardErrorMessages = new SettingInfo
+        public static readonly SettingInfo ForwardErrorMessages = new SettingInfo
         {
             Name = "ServiceControl/ForwardErrorMessages",
             SupportedFrom = new Version(1, 11, 2)
```

Comment:
I wondered if we could remove all `SupportedFrom` that are smaller than version 4 because technically, you always have to upgrade instances from one major to another. The only thing where I think we can't do it is for the maintenance mode because technically, you could have an ancient instance in a newer version of SCMU that you want to put into maintenance mode right?

Reply:
I think that's true.

Reply:
@tmasternak @seanfarmar @mikeminutillo Any thoughts on the above? Should I do a cleanup PR?